agent-inject/agent/agent.go

```diff
@@ -474,10 +479,16 @@ func (a *Agent) Patch() ([]byte, error) {
 
 	//Add Volume Mounts
 	for i, container := range a.Pod.Spec.Containers {
-		a.Patches = append(a.Patches, addVolumeMounts(
-			container.VolumeMounts,
-			a.ContainerVolumeMounts(),
-			fmt.Sprintf("/spec/containers/%d/volumeMounts", i))...)
+		for _, k := range a.Containers {
+			name := strings.Split(k, ",")
+			if strutil.StrListContains(name, container.Name){
+				a.Patches = append(a.Patches, addVolumeMounts(
+					container.VolumeMounts,
+					a.ContainerVolumeMounts(),
+					fmt.Sprintf("/spec/containers/%d/volumeMounts", i))...)
+			}
+		}
```

Comment:
Since `a.Containers` is already a []string, we should be able to simplify this a bit. Something like this?

```suggestion
		if strutil.StrListContains(a.Containers, container.Name){
			a.Patches = append(a.Patches, addVolumeMounts(
				container.VolumeMounts,
				a.ContainerVolumeMounts(),
				fmt.Sprintf("/spec/containers/%d/volumeMounts", i))...)
		}
```